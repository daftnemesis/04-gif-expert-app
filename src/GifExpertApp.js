import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

  // const handleAdd = () => {
  //   //para agregar elementos a un estado, se usa el spread operator
  //   // setCategories([...categories, 'Mazinger Z'])

  //   //o con un callback
  //   setCategories(cats => [...cats, 'Mazinger Z'])
  // }

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory />
      <hr />

      <ol>
        {categories.map((category) => (
          <li key={category}>{category}</li>
        ))}
      </ol>
    </>
  );
};
