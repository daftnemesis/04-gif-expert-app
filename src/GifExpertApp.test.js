import { shallow } from 'enzyme';
import { GifExpertApp } from './GifExpertApp';


describe('Haciendo pruebas en el componente principal', () => { 

  test('haciendo match con el snapshot', () => {

    const wrapper = shallow( <GifExpertApp /> );

    expect( wrapper ).toMatchSnapshot();
    
  });

});