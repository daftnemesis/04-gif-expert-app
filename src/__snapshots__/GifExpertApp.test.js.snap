// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas en el componente principal haciendo match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
